### PR TITLE
Several changes

### PR DESCRIPTION
**Pull request by AkazaRenn**
Did a few things in this change:
* Migrate settings to `json`-based
* https://github.com/popsUlfr/SDH-PauseGames/issues/13 Clear `stickyPause` state on manual resume
* https://github.com/popsUlfr/SDH-PauseGames/issues/22 Add `noAutoPauseSet`
* https://github.com/popsUlfr/SDH-PauseGames/issues/19 Patch `SteamClient.Apps.TerminateApp()` to un-pause games on exit

### CODE REVIEW

**Comment by AkazaRenn** [comment]
There's a problem with the QAM settings, changing it back to draft for now

File: src/index.tsx [hunk 1]
```diff
@@ -200,8 +209,36 @@ const Content: VFC<{ serverAPI: ServerAPI }> = ({}) => {
...
+}
+
+function restoreTerminateApp() {
+  if ((SteamClient.Apps as any)?.OriginalTerminateApp) {
+    console.log("Restoring SteamClient.Apps.TerminateApp()");
+    SteamClient.Apps.TerminateApp = (SteamClient.Apps as any).OriginalTerminateApp;
+    delete (SteamClient.Apps as any).OriginalTerminateApp;
+  }
+}
+
```
Comment (AkazaRenn) [inline src/index.tsx:237]: This crazy thing is for https://github.com/popsUlfr/SDH-PauseGames/issues/19
I bet the reviewers of the store won't like it so it's up to you... I'm checking on patching as a less intrusive solution but it's really making me scratch my head right now... May take some time before I can make something that works. The best would be that we can get some kind of an exception on patching the `SteamClient` object from them.

Comment (AkazaRenn) [reply src/index.tsx:237]: Update: I got the approval from the reviewers in Discord, but I need to use the API provided by Decky Loader to do the override, I'll be looking into it.

**Comment by wynn1212** [comment]
Hi,

Sorry for the late reply, as I've been busy these past months.

The updates look good to me. I can merge this pull request for you, but the final decision on updates and takeover remains with @popsUlfr (as I informed @popsUlfr that my intention was only to update the plugin to fix the DFL problem, not to take over yet).

Here are some problems found during testing:
- The pause/play button is not toggle-able via touchscreen and cursor, only by using the controller or Enter key.
- https://github.com/wynn1212/SDH-PauseGames/blob/b8a35d8dad66289ff96e424011bff33c5ebfff11/src/index.tsx#L190-L198 The description needs to be rewritten due to the changes you made to how the game is paused.

Here are some suggestions:
- Make the pause/play button bigger to make it easier for users to touch.

Also, I just wondering is it possible to migrate this plugin to the new decky API?
- https://wiki.deckbrew.xyz/en/plugin-dev/new-api-migration

Thank you!

**Comment by AkazaRenn** [comment]
Thank you for checking it out! I can definitely update it by following your comments. As for releasing a new version, I have no idea what the best way is either. I've created a PR to the repo of the original author as well but haven't gotten any replies yet. If I'm not able to get the permission for an extended period of time, I may release my version at a new extension if the store agrees.

**Comment by wynn1212** [comment]
No worries. @popsUlfr will eventually respond your replies.

Once you fix the following problems, I will merge your pull request and ask @popsUlfr if we can take over this project.

Also I'm wondering is it possible to fix this issue too?
- https://github.com/popsUlfr/SDH-PauseGames/issues/17 The problem is that you have to manually unpause non-Steam games when launching a non-Steam game while another game is running with Pause in focus loss enabled.
- If I remember correctly, this issue arose after a specific Steam Deck client update changed the behavior of how non-Steam games are launched. Previously, launching a non-Steam game would switch to it, but after the update, launching a non-Steam game while another game is running causes it to launch in the background. This results in the Pause Games plugin automatically pausing the non-Steam game, which doesn't unpause itself after switching to the non-Steam game, requiring manual unpausing.

**Comment by AkazaRenn** [comment]
Sure I can definitely look into the issue you shared, if I find a solution I'll submit it in another PR so I don't make this one any larger.

As for the new Decky API, I checked with the loader devs and they said it is only available on loader v3.0.0+, this version is still in beta and it might be too early to start the migration. Also I am experiencing some problems playing with the new way to call backend functions, I'll need to spend some more time testing it further.

Anyways, I'll submit a PR soon to fix the documents and the pause button, then we can consider the others as the next step.

**Comment by wynn1212** [comment]
Hi there,
Sure, I'll merge this pull request for you as it does not have any major breakage for current testing. And I'll looking for your new pull request.

**Pull request merged**